### PR TITLE
rkt: Add support for termination message, termination reason.

### PR DESCRIPTION
**Pull request by yifan-gu**
If 'TerminationMessagePath' in container spec is set, then
We will mount the termination message log into the container.

Also in GetPodStatus, if the container exits and the 'TerminationMessagePath'
is set, then the 'message' field in container state will be populated.

cc @yujuhong @Random-Liu @jonboulle

### CODE REVIEW

**Comment by yifan-gu** [comment]
Ping

**Comment by yujuhong** [comment]
Reviewing.

**Comment by yujuhong** [comment]
LGTM overall with one nit.

File: pkg/kubelet/rkt/rkt.go [hunk 1]
```diff
@@ -78,11 +78,11 @@ const (
...
 	//TODO: remove the creation time annotation once this is closed: https://github.com/coreos/rkt/issues/1789
-	k8sRktCreationTimeAnno  = "rkt.kubernetes.io/created"
-	k8sRktContainerHashAnno = "rkt.kubernetes.io/containerhash"
-	k8sRktRestartCountAnno  = "rkt.kubernetes.io/restartcount"
-
-	dockerPrefix = "docker://"
+	k8sRktCreationTimeAnno     = "rkt.kubernetes.io/created"
+	k8sRktContainerHashAnno    = "rkt.kubernetes.io/containerhash"
+	k8sRktRestartCountAnno     = "rkt.kubernetes.io/restartcount"
+	k8sRktContainerLogPathAnno = "rkt.kubernetes.io/containerlogpath"
```
Comment (yujuhong) [inline pkg/kubelet/rkt/rkt.go]: nit: Termination message is different from the container log. Can we rename this to make it clear?

**Comment by yifan-gu** [comment]
@yujuhong Fixed.

**Comment by yifan-gu** [comment]
Rebased

**Comment by yifan-gu** [comment]
Re-applied LGTM after rebase.

**Pull request merged**